File: src/main/java/de/maggiwuerze/xdccloader/service/ChannelService.java
package de.maggiwuerze.xdccloader.service;

import de.maggiwuerze.xdccloader.model.entity.Channel;
import de.maggiwuerze.xdccloader.persistence.ChannelRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
@Slf4j
public class ChannelService {

	private final ChannelRepository channelRepository;

	public List<Channel> list() {
		return channelRepository.findAll();
	}

	public Channel save(Channel channel) {
		return channelRepository.save(channel);
	}

	public Channel findById(Long channelId) {
		return channelRepository.findById(channelId).orElse(null);
	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/service/ChannelService.kt
package de.maggiwuerze.xdccwebloader.service

import de.maggiwuerze.xdccwebloader.model.entity.Channel
import de.maggiwuerze.xdccwebloader.model.forms.ChannelFormTO
import de.maggiwuerze.xdccwebloader.persistence.ChannelRepository
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Service
import java.util.*

@Service
class ChannelService(val channelRepository: ChannelRepository) {

    fun list(): List<Channel> {
        return channelRepository.findAll()
    }

    fun save(channel: Channel): Channel {
        return channelRepository.save(channel)
    }

    fun findById(channelId: UUID): Channel? {
        return channelRepository.findById(channelId).get()
    }

    fun delete(id: UUID): HttpStatus {
        return findById(id)?.let {
            channelRepository.delete(it)
            HttpStatus.OK
        } ?: HttpStatus.BAD_REQUEST
    }

    fun update(id: UUID, channelFormTO: ChannelFormTO): Channel {
        return findById(id)?.let {
            it.name = channelFormTO.name
            channelRepository.save(it)
        } ?: throw IllegalStateException("Channel with id $id not found.")
    }
}
